packages/userstorage/src/browser/user-storage-service-filesystem.ts

```diff
@@ -18,38 +18,34 @@ import { DisposableCollection, ILogger, Emitter, Event } from '@theia/core/lib/c
 import { UserStorageChangeEvent, UserStorageService } from './user-storage-service';
 import { injectable, inject } from 'inversify';
 import { FileSystemWatcher, FileChangeEvent } from '@theia/filesystem/lib/browser/filesystem-watcher';
+import { EnvVariablesServer } from '@theia/core/lib/common/env-variables';
 import { FileSystem } from '@theia/filesystem/lib/common';
 import URI from '@theia/core/lib/common/uri';
 import { UserStorageUri } from './user-storage-uri';
 
-export const THEIA_USER_STORAGE_FOLDER = '.theia';
-
 @injectable()
 export class UserStorageServiceFilesystemImpl implements UserStorageService {
 
     protected readonly toDispose = new DisposableCollection();
     protected readonly onUserStorageChangedEmitter = new Emitter<UserStorageChangeEvent>();
-    protected readonly userStorageFolder: Promise<URI | undefined>;
+    protected userStorageFolder: Promise<URI | undefined>;
 
     constructor(
         @inject(FileSystem) protected readonly fileSystem: FileSystem,
         @inject(FileSystemWatcher) protected readonly watcher: FileSystemWatcher,
-        @inject(ILogger) protected readonly logger: ILogger
-
+        @inject(ILogger) protected readonly logger: ILogger,
+        @inject(EnvVariablesServer) protected readonly envServer: EnvVariablesServer
     ) {
-        this.userStorageFolder = this.fileSystem.getCurrentUserHome().then(home => {
-            if (home) {
-                const userStorageFolderUri = new URI(home.uri).resolve(THEIA_USER_STORAGE_FOLDER);
-                watcher.watchFileChanges(userStorageFolderUri).then(disposable =>
-                    this.toDispose.push(disposable)
-                );
-                this.toDispose.push(this.watcher.onFilesChanged(changes => this.onDidFilesChanged(changes)));
-                return new URI(home.uri).resolve(THEIA_USER_STORAGE_FOLDER);
-            }
+        this.userStorageFolder = this.envServer.getUserDataFolderPath().then(userDataFolderPath => {
+            const userStorageFolderUri = new URI('file://' + userDataFolderPath);
```

Comment:
This cannot work on Windows if you pass FS paths as a string between FE and BE. See my comment above https://github.com/eclipse-theia/theia/pull/6650/files#r380712131

Reply:
FYI: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#uripath

Reply:
Yes, using concatenation here looks a bit unsafe. But I think we can replace it with:
```typescript
new URI(userDataFolderPath).withScheme('file');
```

Reply:
You can pass URIs instead of FS paths between the FE and the BE, and it's not an issue.

Reply:
But with the approach above it is not an issue too.

Reply:
@mmorhun sending URIs will be aligned with coding guidelines and with others APIs in Theia

Reply:
@akosyakov it is still not clear to me how to implement it. If I change [`getUserDataFolder`](https://github.com/eclipse-theia/theia/pull/6650/files#diff-c44c169d49c75f07500fa7df70c53670R64) return type to URI, then I have to change it in corresponding protocol interface. That means we intend to send an instance of URI via RPC which is wrong (as object will be corrupted).

Reply:
@mmorhun, the return type remains the same: `string`, but instead of passing an FS path (as a  `string`) we want to send a Theia `URI` `string` (`URI.toString`). Does this clarify your question?

Reply:
@kittaakos yes, thank you.